docker/main/docker-build.yml

```diff
@@ -98,6 +101,7 @@ services:
       context: ./ngen
       target: partitioner_image
       args:
+        NUMPY_CONSTRAINT: ${NUMPY_INSTALL_CONSTRAINT:?Please set global NUMPY_INSTALL_CONSTRAINT in environment config}
```

Comment:
This is an abuse of the args parameter. Do not use this to inject python versions. Stick to setup.py and requirements.txt.